lib/publish.js

```diff
@@ -1,30 +1,86 @@
+const {parse, resolve} = require('path');
+const {stat, readFile} = require('fs-extra');
 const urlJoin = require('url-join');
 const got = require('got');
 const debug = require('debug')('semantic-release:gitlab');
 const resolveConfig = require('./resolve-config');
 const getRepoId = require('./get-repo-id');
+const getAssets = require('./get-assets');
 
 module.exports = async (pluginConfig, context) => {
   const {
+    cwd,
     options: {repositoryUrl},
     nextRelease: {gitTag, gitHead, notes},
     logger,
   } = context;
-  const {gitlabToken, gitlabUrl, gitlabApiPathPrefix} = resolveConfig(pluginConfig, context);
+  const {gitlabToken, gitlabUrl, gitlabApiPathPrefix, assets} = resolveConfig(pluginConfig, context);
   const repoId = getRepoId(gitlabUrl, repositoryUrl);
   const encodedRepoId = encodeURIComponent(repoId);
-  const apiUrl = urlJoin(gitlabUrl, gitlabApiPathPrefix);
   const encodedGitTag = encodeURIComponent(gitTag);
+  const apiBase = urlJoin(gitlabUrl, gitlabApiPathPrefix);
+  const apiProject = urlJoin(apiBase, `/projects/${encodedRepoId}`);
+  const apiOptions = {
+    json: true,
+    headers: {'PRIVATE-TOKEN': gitlabToken},
+  };
+  let finalNotes = notes;
 
   debug('repoId: %o', repoId);
   debug('release name: %o', gitTag);
   debug('release ref: %o', gitHead);
 
+  if (assets && assets.length > 0) {
+    const globbedAssets = await getAssets(context, assets);
+    const assetsMd = [];
+    debug('globed assets: %o', globbedAssets);
+
+    await Promise.all(
+      globbedAssets.map(async asset => {
+        const filePath = asset.path;
+        let fileStat;
+        let file;
+
+        try {
+          file = resolve(cwd, filePath);
+          fileStat = await stat(file);
+        } catch (error) {
+          logger.error('The asset %s cannot be read, and will be ignored.', filePath);
+          return;
+        }
+        if (!fileStat || !fileStat.isFile()) {
+          logger.error('The asset %s is not a file, and will be ignored.', filePath);
+          return;
+        }
+
+        const fileName = asset.name;
+        debug('file path: %o', filePath);
+        debug('file name: %o', fileName);
+
+        // Uploaded assets to the project
+        const {body} = await got.post(urlJoin(apiProject, '/uploads'), {
+          ...apiOptions,
+          body: {file: await readFile(file)},
+        });
+
+        // Replace the markdown asset definition label
+        if (asset.label) {
+          const regExpName = new RegExp(`\\[(${parse(fileName).name}|${fileName})\\]`);
+          body.markdown = body.markdown.replace(regExpName, `[${asset.label}]`);
```

Comment:
This code is not safe. If the API response change in the future this might break. Instead of replacing the label with a regex you should generate the markdown your self with the URL from `body.url`.

Reply:
done https://github.com/semantic-release/gitlab/pull/55/commits/c5403323c7e9c0b1f4c173a7e7018e010b5f4a26